### ISSUE TITLE
pdf not showing 

### ISSUE THREAD

**Issue by trbtl** [lfoppiano/streamlit-pdf-viewer#37]
I’m creating a simple pdf viewer. See the code below. When I click on a button the first time nothing is displayed. When I click on it a second time thh pdf is displayed. What am I doing wrong?

```python
import streamlit as st
from streamlit_pdf_viewer import pdf_viewer
import os

if ‘selected_file’ not in st.session_state:
st.session_state[‘selected_file’] = None

with st.sidebar:
library = st.container()
with library:

    def get_files(directory):
        return [f for f in os.listdir(directory) if os.path.isfile(os.path.join(directory, f))]

    def file_selector(folder_path='.'):
        st.write("Select a document from the list below:")
        files = get_files(folder_path)
        for file in files:
            if st.button(file): 
                st.session_state['selected_file'] = file  

    folder_path = './docs' 
    file_selector(folder_path)
if st.session_state[‘selected_file’] is not None:
filename = st.session_state[‘selected_file’]
st.write(filename)
folder_path = ‘./docs’
uploaded = os.path.join(folder_path, filename)
pdf_viewer(input=uploaded, width=1000)
```

**Comment by trbtl** [comment]
Also, if I select <Rerun> from the menu it will display the PDF.

**Comment by trbtl** [comment]
To check my work I set up in a new folder.  I installed a fresh virtual environment and pip installed a fresh streamlit and streamlit_pdf_viewer.  I then coded a minimumal version of a pdf viewer (see below).  It still won't show the pdf until I select Rerun from the menu it will display the PDF.

import streamlit as st
from streamlit_pdf_viewer import pdf_viewer
import os

filename = 'Choosing a Treatment for Kidney Failure - NIDDK.pdf'
st.write(filename)
folder_path = './docs'
uploaded = os.path.join(folder_path, filename)
pdf_viewer(input=uploaded, width=1000)

**Comment by lfoppiano** [comment]
@trbtl thanks for reporting this issue. It seems related to #35 but I'm surprised that it happens as you pass the `width` parameter. 

Which browser and SO are you using?

**Comment by trbtl** [comment]
Chrome

Thanks
Bob

**Comment by lfoppiano** [comment]
This is likely a regression from #28. If you revert to version 0.0.7 it should work, however, the application will use more CPU because of an infinite loop in the frontend.

**Comment by trbtl** [comment]
That works!  What is it about version 0.0.8 that is a problem with chromium browsers?  Will this be an issue going forward?

Thanks
Bob

**Comment by lfoppiano** [comment]
@trbtl Unfortunately, the issue is not related to the browser, it's a regression from #28 which we did not notice in our demo tests. 
We are working already on a solution, and we will have integration tests trying to spot those situations in future. Leave this issue open and we will update you when a new release will solve this issue.

**Comment by trbtl** [comment]
Thank you so much!  I don’t think I’ve ever had such quick and helpful response.  Keep up the good work.

Thanks
Bob

**Comment by trbtl** [comment]
Now the problem is a problem with displaying a pdf in Streamlit Cloud. The pdf displays correctly when I run it locally, but no luck with displaying it on Streamlit Cloud.
I opened an issue  #43 - https://github.com/lfoppiano/streamlit-pdf-viewer/issues/43
Any ideas?

Thanks
Bob


From: Bob Cole ***@***.***>
Sent: Sunday, February 25, 2024 4:35 PM
To: lfoppiano/streamlit-pdf-viewer ***@***.***>; lfoppiano/streamlit-pdf-viewer ***@***.***>
Cc: mail ***@***.***>; Mention ***@***.***>
Subject: RE: [lfoppiano/streamlit-pdf-viewer] pdf not showing (Issue #37)

That works!  What is it about version 0.0.8 that is a problem with chromium browsers?  Will this be an issue going forward?

Thanks
Bob

**Comment by lfoppiano** [comment]
I just published version 0.0.9 which should solve the issues you've reported. 

At your earliest convenience, could you give it a try and let me know?
